src/docker/shared.rs

```diff
@@ -496,19 +504,60 @@ pub(crate) fn docker_envvars(
     Ok(())
 }
 
+fn mount_to_ignore_cargo_config(
+    docker: &mut Command,
+    paths: &DockerPaths,
+    root: &str,
+    cwd: &str,
+    ignore_cargo_config: bool,
+) -> Result<()> {
+    let is_cargo_home_parent = paths
+        .cargo_home()
+        .parent()
+        .map(|x| x == paths.host_root())
+        .unwrap_or_default();
+    if ignore_cargo_config && !is_cargo_home_parent {
+        let root_path = Path::new(root);
+        let cwd_path = Path::new(cwd);
+        docker.args(&["-v", &cwd_path.join(".cargo").as_posix()?]);
+        let mut relpath = Path::new(cwd).strip_prefix(root_path).wrap_err_with(|| {
+            eyre::eyre!("cwd \"{cwd}\" must be a subdirectory of root \"{root}\"")
+        })?;
+
+        while let Some(parent) = relpath.parent() {
+            let path = root_path.join(parent);
+            docker.args(&["-v", &path.join(".cargo").as_posix()?]);
+            relpath = parent;
+        }
+    }
```

Comment:
doesnt this mount more than it needs?

Reply:
There's still some permission issues here I need to fix, however.

Reply:
But we don't override $HOME/.cargo. We only go up to when `relpath == "/"`, which is the current project directory or CWD. `root` is the project root or CWD. You can test this in a workspace:

```bash
$ cd test-workspace\workspace\binary
$ cross run --target arm-unknown-linux-gnueabihf --verbose
+ "C:\Program Files\Docker\Docker\resources\bin\docker.EXE" run --userns host -e 'PKG_CONFIG_ALLOW_CROSS=1' -e 'XARGO_HOME=/xargo' -e 'CARGO_HOME=/cargo' -e 'CARGO_TARGET_DIR=/target' -e 'CROSS_RUNNER=' -e TERM -e 'USER=ahusz' -v 'C:\Users\ahusz\git\test-workspace\external\external_lib:/mnt/c/Users/ahusz/git/test-workspace/external/external_lib' --rm --user 1000:1000 -v 'C:\Users\ahusz\.xargo:/xargo:Z' -v 'C:\Users\ahusz\.cargo:/cargo:Z' -v /cargo/bin -v 'C:\Users\ahusz\git\test-workspace\workspace:/mnt/c/Users/ahusz/git/test-workspace/workspace:Z' -v 'C:\Users\ahusz\.rustup\toolchains\stable-x86_64-unknown-linux-gnu:/rust:Z,ro' -v 'C:\Users\ahusz\git\test-workspace\workspace\target:/target:Z' -w /mnt/c/Users/ahusz/git/test-workspace/workspace/binary -v /mnt/c/Users/ahusz/git/test-workspace/workspace/binary/.cargo -v /mnt/c/Users/ahusz/git/test-workspace/workspace/.cargo -i -t ghcr.io/cross-rs/arm-unknown-linux-gnueabihf:main sh -c 'PATH=$PATH:/rust/bin cargo run --target arm-unknown-linux-gnueabihf --verbose'
```

Or, abridged for clarity: we only prevent `/mnt/c/Users/ahusz/git/test-workspace/workspace/binary/.cargo` and `/mnt/c/Users/ahusz/git/test-workspace/workspace/.cargo` from being mounted.